src/main/kotlin/com/sourcegraph/cody/edit/widget/LensWidgetGroup.kt

```diff
@@ -240,7 +290,25 @@ class LensWidgetGroup(val session: FixupSession, parentComponent: Editor) :
     }
   }
 
+  private fun <T> onEventThread(handler: Supplier<T>): @NotNull CompletableFuture<T> {
```

Comment:
Maybe it would make sense to put this in some utility class?
It seems to be generally useful.